vars/RedHat.yml

```diff
@@ -41,3 +41,10 @@ __ha_cluster_services:
   - corosync
   - corosync-qdevice
   - pacemaker
+
+# fence agent list moved from defaults
+ha_cluster_fence_agent_packages: "{{
```

Comment:
@tomjelinek was the reason that this was originally in defaults/main.yml because users might want to set this?  If so, we'll have to figure out how to do this in defaults/main.yml, and figure out how to set it to a default value defined in vars/RedHat.yml

Reply:
Would the following in `defaults/main.yml` work?

```yaml
var1_packages_list: "{{ var2_packages_dict[ansible_os_family] }}"

var2_packages_dict:
  RedHat:
    - value1
    - value2
    - value3
  Suse:
    - value1
    - value2
    - value3
```

Reply:
Something like that, yes

Reply:
> @tomjelinek was the reason that this was originally in defaults/main.yml because users might want to set this?

Yes, it was. See https://github.com/linux-system-roles/ha_cluster#ha_cluster_fence_agent_packages

Reply:
> Can we keep ha_cluster_fence_agent_packages as user input defined in defaults as [] and change actual variable to private **__ha_cluster_fence_agent_packages** which is defined from os_family vars depending if ha_cluster_fence_agent_packages | length > 0, otherwise use user input from ha_cluster_fence_agent_packages ?

yes, sounds good

Reply:
change in coming commit, bit changed but simplified. User input retained while os_family vars will kick in if it is empty from defaults.

```
# fence agent list for os_family
__ha_cluster_fence_agent_packages: "{{
    ['fence-agents-all']
    +
    (['fence-virt'] if ansible_architecture == 'x86_64' else [])
  }}"

# fence agent assignment based on user input
ha_cluster_fence_agent_packages: "{{ ha_cluster_fence_agent_packages if
 ha_cluster_fence_agent_packages | length > 0
 else __ha_cluster_fence_agent_packages }}"
```

Reply:
ok

Reply:
Fixed in current commit